File: app-java/src/main/java/com/paytpv/appjava/utils/PurchaseHandler.java
package com.paytpv.appjava.utils;

import com.paytpv.androidsdk.Model.PTPVRequests.Purchase.PTPVPurchaseDetails;

import java.util.ArrayList;

public class PurchaseHandler {
    private static PurchaseHandler singleton = new PurchaseHandler( );
    private ArrayList<PTPVPurchaseDetails> purchaseArray = new ArrayList<>();

    private PurchaseHandler() { }

    public static PurchaseHandler getInstance( ) {
        return singleton;
    }

    public ArrayList<PTPVPurchaseDetails> getPurchaseArray() {
        return purchaseArray;
    }
}

File: app-kotlin/src/main/java/com/paytpv/paytpv/utils/PurchaseHandler.kt
package com.paytpv.paytpv.utils

import com.paytpv.androidsdk.Model.PTPVRequests.Purchase.PTPVPurchaseDetails

class PurchaseHandler private constructor() {

    private object Holder {
        val INSTANCE = PurchaseHandler()
    }

    companion object {
        val instance: com.paytpv.paytpv.utils.PurchaseHandler by lazy { Holder.INSTANCE }
    }

    var purchaseArray: ArrayList<PTPVPurchaseDetails>? = ArrayList()
}